### PR TITLE
[201811] Dell: Fix S6000 reboot issue 

### PR DESCRIPTION
**Pull request by aravindmani-1**
**- Why I did it**
Reboot command in Dell S6000 failed to reboot the switch.
**- How I did it**
Added retry mechanism and CPU reset.
**- How to verify it**
Execute reboot command in Dell S6000.
**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**
[S6000_reboot_UT.txt](https://github.com/Azure/sonic-buildimage/files/5930031/S6000_reboot_UT.txt)

### CODE REVIEW

**Comment by aravindmani-1** [comment]
@sujinmkang , @jleveque can you please review this PR?.

File: device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot [hunk 1]
```diff
@@ -1,13 +1,63 @@
...
-echo 0 > /sys/class/gpio/gpio10/value
+PORT_RES = '/dev/port'
+NVRAM_RES = '/dev/nvram'
+COLD_RESET = 0xE     # Cold Reset
+WARM_RESET = 0x6     # Warm Reset
+RESET_REG = 0xCF9
 
+def io_reg_write(resource, offset, val):
+    fd = os.open(resource, os.O_RDWR)
+    if(fd < 0):
```
Comment (jleveque) [inline device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot:15]: Parentheses unnecessary in Python. Please remove.

File: device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot [hunk 2]
```diff
@@ -1,13 +1,63 @@
...
+        print('file open failed %s" % resource')
+        return
+    if(os.lseek(fd, offset, os.SEEK_SET) != offset):
```
Comment (jleveque) [inline device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot:18]: Parentheses unnecessary in Python. Please remove.

File: device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot [hunk 3]
```diff
@@ -1,13 +1,63 @@
...
+        print('lseek failed on %s' % resource)
+        return
+    ret = os.write(fd, struct.pack('B', val))
+    if(ret != 1):
```
Comment (jleveque) [inline device/dell/x86_64-dell_s6000_s1220-r0/platform_reboot:22]: Parentheses unnecessary in Python. Please remove.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Pull request merged**